### PR TITLE
Follow up to improve gdrive and move auth before service accounts

### PR DESCRIPTION
**Pull request by shcheklein**
Followup based on https://github.com/iterative/dvc.org/pull/3473#issuecomment-1112659711

-----------------------------

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: content/docs/user-guide/setup-google-drive-remote.md [hunk 1]
```diff
@@ -163,56 +163,14 @@ $ dvc remote modify myremote gdrive_client_secret 'client-secret'
...
-![](/img/gdrive-share-with-service-account.png)
-
 ## Authorization
 
+There are two ways to give DVC access to a GDrive directory or a disk. The one
+descibed in this section gives DVC a way to access GDrive on behalf of a user
+account. It's the simplest way to use, and it's fine to use to run DVC locally.
+If some automation is needed (e.g. CI/CD) we recommend to
+[use a service account](#using-service-accounts) instead.
+
```
Comment (shcheklein) [inline content/docs/user-guide/setup-google-drive-remote.md:173]: One extra paragraph above

File: content/docs/user-guide/setup-google-drive-remote.md [hunk 2]
```diff
@@ -254,3 +212,43 @@ authentication process. If `GDRIVE_CREDENTIALS_DATA` is set, the
 `gdrive_user_credentials_file` value (if provided) is ignored.
 
 > Please note our [Privacy Policy (Google APIs)](/doc/user-guide/privacy).
+
+## Using service accounts
```
Comment (shcheklein) [inline content/docs/user-guide/setup-google-drive-remote.md:216]: This section is unchanged and only moved to the bottom

File: content/docs/user-guide/setup-google-drive-remote.md [hunk 3]
```diff
@@ -163,56 +163,14 @@ $ dvc remote modify myremote gdrive_client_secret 'client-secret'
...
-
-![](/img/gdrive-share-with-service-account.png)
-
 ## Authorization
 
+There are two ways to give DVC access to a GDrive directory or a disk. The one
+descibed in this section gives DVC a way to access GDrive on behalf of a user
+account. It's the simplest way to use, and it's fine to use to run DVC locally.
+If some automation is needed (e.g. CI/CD) we recommend to
+[use a service account](#using-service-accounts) instead.
```
Comment (jorgeorpinel) [inline content/docs/user-guide/setup-google-drive-remote.md:172]:
```suggestion
<admon type="info">

This covers simple authentication, which gives DVC access to GDrive
on behalf of a user account. This is ideal to use to run DVC locally, for example.
If some automation is needed (e.g. CI/CD) we recommend
[using a service account](#using-service-accounts) instead.

</admon>
```

File: content/docs/user-guide/setup-google-drive-remote.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/setup-google-drive-remote.md:216]: Not exactly unchanged: It's missing the `GDRIVE_CREDENTIALS_DATA` paragraph, but that's already in **Authentication** which is now before this section. Is this the intention? (To have it only above)

**Review by jorgeorpinel — APPROVED** [review]

**Comment by dberenbaum** [comment]
Looking at it again, I think "Authorization" should even be above "Using a custom Google Cloud Project" (sorry for not catching this earlier). So the structure would be:

1. Quick start
2. URL format - provides more detail on the quick start scenario
3. Authorization - provides more detail on the quick start scenario and introduces auth options for advanced scenarios below
4. Using a custom Google Cloud Project - explains advanced best practice that is needed to use service accounts
5. Using service accounts - explains advanced best practice for automation

Thoughts?

Happy to make the changes myself if it's easier once we all agree.

File: content/docs/user-guide/setup-google-drive-remote.md (hunk 2 above)
Comment (dberenbaum) [reply content/docs/user-guide/setup-google-drive-remote.md:219]: Right, especially with the new paragraph about specifying that the "Authorization" section is about user accounts, it seems unclear to me that `GDRIVE_CREDENTIALS_DATA` can also be used for service accounts.

Comment (shcheklein) [reply content/docs/user-guide/setup-google-drive-remote.md:219]: > Not exactly unchanged:

good catch, that's a bug, I'll fix it. It's not intentional at all.

**Comment by shcheklein** [comment]
> Looking at it again, I think "Authorization" should even be above "Using a custom Google Cloud Project"

I think that was intentional, so that it's not considered as _advanced_ but rather as recommended (in most cases you would want to have an independent app so that your rate limits don't overlap with other DVC users).

Also, it's an optional configuration step, while Auth follows both of them.

So, I don't see changing the order as a fixing something step, but I'm fine with that if you guys decide to this.

**Comment by dberenbaum** [comment]
Sounds good. I'm not sure the GCP -> Auth -> Service Accounts order is any better than the current GCP -> Service Accounts -> Auth order, so I'm open to closing, merging, or reordering as Auth -> GCP -> Service Accounts. @jorgeorpinel If you have any opinion, happy to go with that.

**Comment by jorgeorpinel** [comment]
Sounds like service accounts are "more advanced" so it makes sense to have them last. If it's the recommended method then maybe we should be more explicit about that and move them back up. But it's probably not worth going back to it now... We have to maintain this periodically anyway so I'm sure we'll have another chance in the future.

> TBH ideally I'd prefer if we didn't have to be so specific with 3rd party tool docs.

**Pull request merged**